tracer/src/Datadog.Trace/Configuration/ExporterSettings.cs

```diff
@@ -177,142 +199,180 @@ public int PartialFlushMinSpans
+            {
+                if (TrySetAgentUriAndTransport(agentUri))
+                {
+                    return;
+                }
+            }
 
-            // Agent port is set to zero in places like AAS where it's needed to prevent port conflict
-            // The agent will fail to start if it can not bind a port, so we need to override 8126 to prevent port conflict
-            // Port 0 means it will pick some random available port
-            var hasExplicitHostOrPortSettings = (agentPort != null && agentPort != 0) || agentHost != null;
+            if (!string.IsNullOrWhiteSpace(tracesPipeName))
+            {
+                TracesTransport = TracesTransportType.WindowsNamedPipe;
+                TracesPipeName = tracesPipeName;
+                SetAgentUri(agentHost ?? DefaultAgentHost, agentPort ?? DefaultAgentPort); // this one can throw
+                return;
+            }
 
-            if (hasExplicitHostOrPortSettings)
+            // This property shouldn't have been introduced. We need to remove it as part of 3.0
+            // But while it's here, we need to handle it properly
+            if (!string.IsNullOrWhiteSpace(tracesUnixDomainSocketPath))
             {
-                if (agentHost?.StartsWith(UnixDomainSocketPrefix) ?? false)
+                if (TrySetAgentUriAndTransport(UnixDomainSocketPrefix + tracesUnixDomainSocketPath))
                 {
-                    traceTransport = TracesTransportType.UnixDomainSocket;
-                    TracesUnixDomainSocketPath = agentHost;
+                    return;
                 }
-                else
+            }
+
+            if ((agentPort != null && agentPort != 0) || agentHost != null)
+            {
+                // Agent port is set to zero in places like AAS where it's needed to prevent port conflict
+                // The agent will fail to start if it can not bind a port, so we need to override 8126 to prevent port conflict
+                // Port 0 means it will pick some random available port
+
+                if (TrySetAgentUriAndTransport(agentHost ?? DefaultAgentHost, agentPort ?? DefaultAgentPort))
                 {
-                    // The agent host is explicitly configured, we should assume UDP for metrics
-                    forceMetricsOverUdp = true;
-                    traceTransport = TracesTransportType.Default;
+                    return;
                 }
             }
-            else if (!string.IsNullOrWhiteSpace(TracesPipeName))
+
+            if (_fileExists(DefaultTracesUnixDomainSocket))
             {
-                traceTransport = TracesTransportType.WindowsNamedPipe;
-
-                TracesPipeTimeoutMs = source?.GetInt32(ConfigurationKeys.TracesPipeTimeoutMs)
-#if DEBUG
-                    ?? 20_000;
-#else
-                    ?? 500;
-#endif
+                TrySetAgentUriAndTransport(UnixDomainSocketPrefix + DefaultTracesUnixDomainSocket);
```

Comment:
This seems unnecessary, as it then tries to create a URL, then strips this prefix off, checks if the file exists again etc. I think we can just do
```suggestion
                SetUdsAsTraceTransport(DefaultTracesUnixDomainSocket);
```

Reply:
Actually, this is a breaking change. We used to set the Uri not matter what. So I need to set it again to avoid a potential, though improbable NullRef.
If I want to have the exact same behaviour as before, I will set it as the default http value. Even though it doesn't make sense.

Reply:
Reintroduced in f70a9ecc248021432f7325c95c6622f39972140d

Reply:
> If I want to have the exact same behaviour as before, I will set it as the default http value. Even though it doesn't make sense.

I'd argue we _don't_ want the exact same behaviour as before, but I don't think it really matters.